### PR TITLE
Fix URL redirection issue - fixes #21

### LINKED ISSUE #21: Breaks redirection URLs
The plugin break Twitter's redirection (those URLs found in Twitter emails). Ie. https://twitter.com/i/redirect?url=<originalURL> will be rewritten into https://mobile.twitter.com/i/redirect?url=<originalURL>, which is not available.

### LINKED ISSUE #24: Share links broken
The youtube twitter share button for example is broken, when using your app. 

It does the redirect, but then results in a 404.

### PR DESCRIPTION
**Pull request by phanan**

### CODE REVIEW

File: extension/background.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+
 chrome.webRequest.onBeforeRequest.addListener(details => {
 	if (details.method !== 'GET') {
 		return;
 	}
 
-	// if the URL is for an intent, don't change the URL
-	if (/twitter.com\/intent\//.test(details.url)) {
-		return;
+	for (let re of exceptionRE) {
```
Comment (paulmolluzzo) [inline extension/background.js]: shouldn't this `let` be a `const`? I get a warning in the tests.

**Comment by paulmolluzzo** [comment]
Other than the minor comment about `let` being a `const` I say LGTM.

// @sindresorhus

File: extension/background.js (hunk 1 above)
Comment (phanan) [inline extension/background.js]: My bad. Btw the same warning is triggered on https://github.com/sindresorhus/refined-twitter/blob/master/extension%2Fbackground.js#L30 as well.

Comment (paulmolluzzo) [inline extension/background.js]: Uh, maybe we have to go back to `let`...

https://github.com/sindresorhus/refined-twitter/pull/4#issuecomment-216825742

@sindresorhus what's the plan for FF support? If we can't use `const` in these loops we should document that and maybe modify the xo config or ignore these lines. Thoughts on that? The warnings might pile up.

Comment (phanan) [inline extension/background.js]: Oh LOL.

**Comment by sindresorhus** [comment]
I now wonder if it would make more sense to whitelist URLs to redirect to the mobile site rather than blacklisting? There might be even more exceptions needed that we aren't seeing and with whitelisting it's clear what URLs are supported for the mobile site.

File: extension/background.js (hunk 1 above)
Comment (sindresorhus) [inline extension/background.js]: Fixed: https://github.com/sindresorhus/refined-twitter/commit/bbbf1c52ede0a37e3134026d27a34f3146e03ffd

Firefox will fix it soon, so I've just turned the warning off for now: https://github.com/sindresorhus/refined-twitter/commit/bbbf1c52ede0a37e3134026d27a34f3146e03ffd

**Comment by phanan** [comment]
Not against whitelisting, but it kind of defeats the idea behind the extension doesn't it? It sounds like "Hey look, we have this extension that basically redirects all Twitter links to their mobile counterparts" and then in the code "No, not really, only these links work, sorry."

**Comment by sindresorhus** [comment]
Alright, we can try it this way for now.

**Comment by sindresorhus** [comment]
Thanks @phanan :)

**Comment by phanan** [comment]
Phở 👍

File: source/features/code-highlight.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			return key;
+		}
+	}
+}
+
+function string2dom(domString) {
+	const dom = document.createElement('div');
+	dom.innerHTML = domString;
+	return dom;
+}
```
Comment (sindresorhus) [inline source/features/code-highlight.js]: Can you move this into https://github.com/sindresorhus/refined-twitter/blob/master/source/libs/utils.js and call it `domify`? You can use the source from https://github.com/sindresorhus/refined-github/blob/master/source/libs/domify.js

File: webpack.config.js [hunk 3]
```diff
@@ -14,7 +14,10 @@ module.exports = {
 		new CopyWebpackPlugin([{
 			from: '*',
 			context: 'source',
-			ignore: '*.js'
+			ignore: '*.js$'
```
Comment (sindresorhus) [inline webpack.config.js]: Why this? It's a glob, not regex.

**Pull request merged**